## Assignments on variables expressions and statements

### 1. Take the input from the user for(Total number of people,Number of seats for bus. Based on two inputs Decide how many number of buses required

In [1]:
people = int(input("number of people:"))

seats = int(input("number of seats:"))

buses = (people+seats-1)//seats

print(f"Number of buses: {buses}")
 

number of people: 50
number of seats: 4


Number of buses: 13


### 2.take temperature from the user and convert foreign heat -> Celsius.

In [2]:
Fahrenheit = float(input("Enter Fahrenheit Temp: "))

Celsius = (Fahrenheit - 32) * 5/9

print(f"Enter the Celsius temp is {int(Celsius)}")

 

Enter Fahrenheit Temp:  89


Enter the Celsius temp is 31


### 3.take temperature from the user and convert Celsius → foreign heat.

In [3]:
Celsius = float(input("Enter Celsius Temp: "))

Fahrenheit = (Celsius * 9/5) + 32

print(f"Enter the Fahrenheit temp is {int(Fahrenheit)}")
 

Enter Celsius Temp:  90


Enter the Fahrenheit temp is 194


### 5.Take the distance in km

### Show that in cm, meters, in milli meters, cents, feets, yards

In [4]:
Kilometers = float(input("Enter no. of Kilometers: "))

centimeters = Kilometers * 100000 

meters = Kilometers * 1000 

millimeters = Kilometers * 1000000

cents = Kilometers * 39370.1

feets = Kilometers * 3280.84

yards = Kilometers * 1093.61

print(f" Kilometers: {centimeters: .2f}")  

print(f" Kilometers: {meters: .2f}")  

print(f" Kilometers: {millimeters: .2f}")  

print(f" Kilometers: {cents: .2f}")  

print(f" Kilometers: {feets: .2f}")  

print(f" Kilometers: {yards: .2f}")
 

Enter no. of Kilometers:  4


 Kilometers:  400000.00
 Kilometers:  4000.00
 Kilometers:  4000000.00
 Kilometers:  157480.40
 Kilometers:  13123.36
 Kilometers:  4374.44


### 6.Take the size of your hard disk in GB  Show that in MB, KB, TB, PB



In [5]:
disk_size = float(input("Enter size in GB:- "))
print(f'The size in {disk_size}GB to MB is {disk_size*1000}MB')
print(f'The size in {disk_size}GB to KB is {disk_size*1000000}KB')
print(f'The size in {disk_size}GB to TB is {disk_size*0.001}TB')
print(f'The size in {disk_size}GB to PB is {disk_size/1000000}PB')

Enter size in GB:-  1


The size in 1.0GB to MB is 1000.0MB
The size in 1.0GB to KB is 1000000.0KB
The size in 1.0GB to TB is 0.001TB
The size in 1.0GB to PB is 1e-06PB


### 4.take four number from the user (variables name it as x1,x2,x3,x4)

### Do the below operations

### (x1+x2)**2, (x3+x4)**3

### variance

### standard deviation: sqrt(variance): User math module. Math.sqrt(variance)

### Regression

### y=mx+b

### m=1.23

### b=0.045

### find out y

### y=m*(x1+x2+x3+x4)+b

### Find the average of four numbers

### Find the sum of four numbers

In [6]:
import math
x1 = int(input())
x2 = int(input())
x3 = int(input())
x4 = int(input())
res1 = (x1+x2)**2
res2 = (x3+x4)**3
mean = (x1+x2+x3+x4)/4
varience = ((x1-mean)**2 + (x2-mean)**2 + (x3-mean)**2 + (x4-mean)**2)/4
std_dev = math.sqrt(varience)
y = 1.23*(x1+x2+x3+x4)+0.045
avg = mean
sum_of_four = x1+x2+x3+x4
print(res1)
print(res2)
print(varience)
print(std_dev)
print(y)
print(avg)
print(sum_of_four)


 1
 2
 3
 4


9
343
1.25
1.118033988749895
12.345
2.5
10


### 7. Take name, age, height from the user and print like below

### The details of the person: Name:name of the person, Age:age of the person, Height:height of the person

### Note: make sure that no space between : and a value and should be space after “COMA”

In [7]:
name = input()
age= input()
height=input()
print(f'The details of the person: Name:{name}, age:{age}, height:{height}')

 siri
 26
 4.3


The details of the person: Name:siri, age:26, height:4.3


### 8. BMI calculation: take required parameters for BMI calculation from the user and calculate BMI of the person.

In [8]:
weight = float(input())
height = float(input())
bmi = weight/(height*height)
print(f'calculation of bmi:{bmi}')


 23
 4


calculation of bmi:1.4375


### 9. name="Jayaram"

### age=1.6

### height=3.5356234

### weight=10.343856783

### By using above inputs print the output

### Name:Jayaram, Age:1.6, Height:3.54, Weight:10.344

### Note: Use format specifiers(%s, %d, %f)

In [9]:
name = "Jayaram"
age =  1.6
height = 3.5356234
weight = 10.343856783
print(f'Name:%s, Age:%.1f , Height:%.2f, Weight:%.3f'%(name,age,height,weight))


Name:Jayaram, Age:1.6 , Height:3.54, Weight:10.344


### 10. Take three upper case letters from the user convert in to small case.

In [10]:
x="ABC"
#print(f'print small case'{})
print(dir(x))
p = x.lower()
print(p)

['__add__', '__class__', '__contains__', '__delattr__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__getnewargs__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__iter__', '__le__', '__len__', '__lt__', '__mod__', '__mul__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__rmod__', '__rmul__', '__setattr__', '__sizeof__', '__str__', '__subclasshook__', 'capitalize', 'casefold', 'center', 'count', 'encode', 'endswith', 'expandtabs', 'find', 'format', 'format_map', 'index', 'isalnum', 'isalpha', 'isascii', 'isdecimal', 'isdigit', 'isidentifier', 'islower', 'isnumeric', 'isprintable', 'isspace', 'istitle', 'isupper', 'join', 'ljust', 'lower', 'lstrip', 'maketrans', 'partition', 'removeprefix', 'removesuffix', 'replace', 'rfind', 'rindex', 'rjust', 'rpartition', 'rsplit', 'rstrip', 'split', 'splitlines', 'startswith', 'strip', 'swapcase', 'title', 'translate', 'upper', 'zfill']
abc


### 11. take base and exponent value from the user and print like in mathematics:

### example: base=2, exponent=3: 23

### Use: 2\u00b3

In [11]:
base = input('enter base value')
exponent = input('enter exponent value')
print(f'8\u00b3')

enter base value 2
enter exponent value 5


8³


In [ ]:
base = input('Enter base value: ')
exponent = input('Enter exponent value: ')


# Converting exponent to superscript if possible
# Using a dictionary for common exponents
superscripts = {
    '0': '\u2070', '1': '\u00B9', '2': '\u00B2', '3': '\u00B3',
    '4': '\u2074', '5': '\u2075', '6': '\u2076', '7': '\u2077',
    '8': '\u2078', '9': '\u2079'
}

def to_superscript(number):
    return ''.join(superscripts.get(digit, '') for digit in str(number))

# Print base and exponent with superscript
print(f'{base}{to_superscript(exponent)}')


### 12. Take some groceries cost prices and print total cost and average cost, what is the max cost, what is the minimum cost.

In [ ]:
sugar=input('enter the sugar value')
fruits=input('enter the fruits value')
oils=input('enter oil value')
chocolates=input('enter chocolate value')
total=sugar+fruits+oils+chocolates
print(total)
print(int(total)/4)
print(max(sugar,fruits,oils,chocolate))
print(min(sugar,fruits,oils,chocolate))

# Conditional statements

### 13. Take the input from the user for(Total number of people, total number of buses, Number of seats for bus, adjust factor). Based on four inputs
### Decide whether there is sufficient buses or not and give solution for how many extra buses required.

In [5]:
people = int(input("number of people:"))
seats = int(input("number of seats:"))
buses = int(input("number of buses"))
factor = float(input("Enter the adjust factor"))
#seats_per_bus = seats//buses
no_of_buses = (people)//(seats+factor)
print(no_of_buses)
if (people)%(seats+factor) != 0:
    no_of_buses += 1
    
    if buses > no_of_buses:
        print(f"{no_of_buses} buses required but")
        print(f"{buses-no_of_buses} extra buses are their")
    else:
        print("No Sufficient buses")
        print(f"{no_of_buses-buses} buses are required")
        
else:
    print("Sufficient buses")






number of people: 200
number of seats: 20
number of buses 4
Enter the adjust factor 20


5.0
Sufficient buses


### 14.Take number from the user decide whether it is even or odd.


In [15]:
value = int(input("Enter a value"))
if value%2 == 0:
    print("EVEN")
else:
    print("ODD")

Enter a value 66


EVEN


### 15.take number from the user decide whether it is positive number or negative number

In [7]:
num = input()
if num[0] == "-":
    print("Negative number")
else:
    print("Positive number")

 -2


Negative number


### 16.take a string from the user print the length. if the user not given anything then show an error message

In [14]:
string = input("Enter a word")
len_string = len(string)
if len_string
   print(len_string) 
else:"Error"
    

SyntaxError: expected ':' (2733888917.py, line 3)

In [1]:
string = input("Enter a word: ")
len_string = len(string)

if len_string > 0:  # Check if the length of the string is greater than 0
    print(len_string)
else:
    print("Error")  # Print "Error" if the string is empty

Enter a word:  


Error


### 17.code to perform mathematical operations. take two numbers from the user and show the below menu

### 1. add,

### 2. sub,

### 3. mul,

### 4.div,

### 5.quit

### Enter an option:

### 
### based on the option need to perform an operations

In [1]:
a = int(input("Enter 'a' value: "))

b = int(input("Enter 'b' value: "))

print( "1.add\n2.sub\n3.mul\n4.division\n5.quit")

opt = input("Enter an option:")

if opt=="1": 

    print(a+b)

elif opt=="2":  

    print(a-b)

elif opt=="3": 

    print(a*b)

elif opt=="4":  

    print(a/b)

elif opt=="5":  

    print("quit")

else:

    print("enter valid option")

print("Thank you")
 

Enter 'a' value:  1
Enter 'b' value:  5


1.add
2.sub
3.mul
4.division
5.quit


Enter an option: 1


6
Thank you


### 18. show the menu:

### 1. kids

### 2. Men's

### 3. Women's

### Show the corresponding message based on the selection.

### Option:1: you are a kid

### 2: you are a gentlemen

### 3: you are a good women

### Mention error message if the option value >3.

In [2]:
print("Menu:\n1. Kids\n2. Men's\n3. Women's")

option = int(input("Select an option (1, 2, or 3): "))

if option == 1:

    print("You are a kid.")

elif option == 2:

    print("You are a gentleman.")

elif option == 3:

    print("You are a good woman.")

else:

    print("Invalid option. Please select 1, 2, or 3.")

print("Thank you for shopping")
 

Menu:
1. Kids
2. Men's
3. Women's


Select an option (1, 2, or 3):  1


You are a kid.
Thank you for shopping


### 19. write a program to chcek given substring is there in actual string or not?

### example: act="python is a pure object oriented programing language"

### check whether “pure” is there in act or not.

### Note: Use in operator

In [ ]:
print("Operating Systems:\n\t1.Windows\n\t2.Android\n\t3.MAC")
 
option = input("Enter the option(1-3):")
# Checks if option is 1
if option == '1':
    print("Go to first floor and buy windows laptop or mobile")
# Checks if option is 2 
elif option == '2':
    print("Go to second floor and buy android mobiles")
# Checks if option is 3
elif option == '3':
    print("Go to third floor and buy mac laptop or iphones")
else:
    print("There is only three floors, please select 1 or 2 or 3")

### 20. Take three numbers from the user and decide which is big

In [3]:
num1 = input("Enter the first number:")

num2 = input("Enter the second number:")

num3 = input("Enter the third number:")

if float(num1)>float(num2) and float(num1)>float(num3):

    print(f"{num1} is bigger")

elif float(num2)>float(num1) and float(num2)>float(num3):

    print(f"{num2} is bigger")

else:

    print(f"{num3} is bigger")
 

Enter the first number: 10
Enter the second number: 20
Enter the third number: 30


30 is bigger


### 21. Take age and gender from the user and decide whether he is eligible for marriage in India or not.

### Age criteria: men age>24, women>21

In [4]:
age = input("Enter the age:")

gender = input("Enter the gender(male/female):")

# Check eligibility for marriage based on gender and age

if gender == 'male':

    if int(age)>24:

      print("you are eligible for marriage in India")

    else:

      print("You are not eligible for marriage in India")

elif gender == 'female':

    if int(age)>21:

        print("you are eligible for marriage in India")

    else:

        print("You are not eligible for marriage in India")

else:

    print("Invalid gender. Please enter male or female")
 

Enter the age: 24
Enter the gender(male/female): 21


Invalid gender. Please enter male or female


### 22. Take an age and gender from the user: and mention that what he/she can do in india.

### """

### conditions

### 1. Theatre: 5 for men 7 for women

### 2. Voting system: 18 for men and women

### 3. Marriage in india: 23 for men and for women >21

### 4. For govt jobs: (min:18, max:32) for men and (min:18, max:34) for women

### 5. For driving licence: (min:18, max:60) for men and women

### Eligibility:

### 1. theatre

### 2. Voting system

### 3. Marriage in india

### 4. For govt obs

### 5. For driving licence:

### Enter an option:

### Gender:

### 1. men

### 2. women

### Enter an option:

### Enter an age of person:

In [25]:
print("1.Men\n2.Women")

gender = input("enter your gender")
age = input("enter an age:")
age = float(age)
if age>=18:
   print("He can apply for voting")
if 18<= age >= 60:
   print("He can apply for driving licence")
if gender == "1":
     if age>=5:
        print("he can got to movie")
     if age>=23:
        print("He can get married")
     if age>=18 and age<=34:
       print("He can apply for govt gobs")
elif  gender == "2": 
    if age>=7:
         print("she can got to movie")
    if age >= 18:
        print("sHe can apply for voting")
    if age>=23:
        print("sHe can get married")
    if age>=18 and age<=34:
        print("sHe can apply for govt gobs")

1.Men
2.Women


enter your gender 2
enter an age: 65


He can apply for voting
He can apply for driving licence
she can got to movie
sHe can apply for voting
sHe can get married


### 23. operating systems:

### 1.windows

### 2.android

### 3.mac

### Enter an option:

### If the user enters 1 then show "Goto first floor and buy windows laptop or mobile"

### If the user enters 2 then show "Goto second floor and buy adroid mobiles"

### If the user enters 3 then show "Goto third floor and buy mac laptop or iphones"

### If the user enters other than 1 or 2 or 3 then show "There is only three floors, please select 1 or 2 or 3"

In [5]:
print("Operating Systems:\n\t1.Windows\n\t2.Android\n\t3.MAC")
 
option = input("Enter the option(1-3):")

# Checks if option is 1

if option == '1':

    print("Go to first floor and buy windows laptop or mobile")

# Checks if option is 2 

elif option == '2':

    print("Go to second floor and buy android mobiles")

# Checks if option is 3

elif option == '3':

    print("Go to third floor and buy mac laptop or iphones")

else:

    print("There is only three floors, please select 1 or 2 or 3")


Operating Systems:
	1.Windows
	2.Android
	3.MAC


Enter the option(1-3): 2


Go to second floor and buy android mobiles


### 24.Given an age, figure out whether someone's a baby, toddler, child, teenager, adult or old codger.

In [2]:
age = int(input("Enter the age: "))
 
# Determine the category based on age

if age < 2:

    print("Baby")

elif 2 <= age < 4:

    print("Toddler")

elif 4 <= age < 13:

    print("Child")

elif 13 <= age < 20:

    print("Teenager")

elif 20 <= age < 65:

    print("Adult")

else:

    print("Old Codger")
 

Enter the age:  6


Child


### 25.Take two number a,b from the user and check whether a is divisible by b or not

In [1]:
a = input("Enter number a:")
b= input("Enter number b:")
a = float(a)
b = float(b)
if a%b == 0:
    print(f"{a} is divisible by {b}")
else:
    print(f"{a} is not divisible by {b}")

Enter number a: 45
Enter number b: 87


45.0 is not divisible by 87.0


### 26. Take a letter from the user and print that letter belongs to which category I.e is it a small letter or capital letter or number or special symbol

In [33]:
char=input("enter any single letter")
if len(char) != 1:
    print("Please enter only one character.")
else:
    ascii_val = ord(char)
    
    if 65 <= ascii_val <= 90:
        print("The character is an uppercase letter.")
    elif 97 <= ascii_val <= 122:
        print("The character is a lowercase letter.")
    elif 48 <= ascii_val <= 57:
        print("The character is a digit.")
    elif (32 <= ascii_val <= 47) or (58 <= ascii_val <= 64) or \
         (91 <= ascii_val <= 96) or (123 <= ascii_val <= 126):
        print("The character is a special symbol.")
    else:
        print("The character does not fall into any of the recognized categories.")

enter any single letter h


The character is a lowercase letter.


## looping statements

### 25. take a number from the user and check whether it is prime?

In [1]:
n = int(input("Enter a number:- "))

if n <= 1:
    print("Not Prime")
else:
    c = 0
    for i in range(1,n+1):
        if n%i == 0:
            c += 1
    if c == 2:
        print("It's Prime")
    else:
        print("Not Prime")


Enter a number:-  20


Not Prime


### 26. take a string from the user and check contains only digits or not?

In [3]:
string = input("Enter input:- ")

if string.isdigit():
    print("Contain only digits")
else:
    print("Does not contain only digits")

Enter input:-  qwewq234


Does not contain only digits


### 27. take a string from the user and check contains only alphabets or not?

In [6]:
string = input("enter input")
if string.isalpha():
    print("contains only alphabets")
else:
    print("does not contain only alphabets")

enter input hjbih34


does not contain only alphabets


### 28. take a string from the user and check contains only special chars or not?

In [8]:
string = input("enter input")
# if all(not i.isalnum() for i in string):
#     print("The string contain only special characters")
# else:
#     print("The string does not contain special characters")

c = 0
for i in string:
    if not i.isalnum():
        c += 1 
if c==len(string):
    print("Contains only speical characters")
else:
    print("Does not contain only special characters")
    


enter input @#


Contains only speical characters


### 29.take a string from the user and check contains only capital letters or not?

In [9]:
string = input("Enter input:- ")

if all(i.isupper() for i in string):
    print("Contains only capital letters")
else:
    print("Does not contan only capital letters")

Enter input:-  ERDXFCGVHBJN


Contains only capital letters


### 30.take a string from the user and check contains only small letters or not?

In [10]:
string = input("Enter input:- ")

if all(i.islower() for i in string):
    print("Contains only small letters")
else:
    print("Does not contain only small letters")


Enter input:-  vghvbjhkm


Contains only small letters


### 31. WAP to replace last n occurrence of give string.
### For example:”apple,orange,apple,grape,orange,apple,apple,orange”
### source: “apple”
### last occurrences: 2
### replace with: APPLE
### output:”apple,orange,apple,grape,orange,APPLE,APPLE,orange”s: 2

In [11]:
s = "apple,orange,apple,grape,orange,apple,apple,orange"
list_s = s.split(",")
new = list_s[::-1]
c = 0
for i in range (len(list_s[::-1])):
    if new[i] == 'apple':
        new[i] = "APPLE"
        c += 1
        if c == 2:
            break       
        
print(",".join(new[::-1]))   

apple,orange,apple,grape,orange,APPLE,APPLE,orange


### 32. WAP to check given string contains numbers or not. it should consider float numbers also.

In [13]:
string = input("Enter input:- ")
c = 0
for i in string:
    if i.isdigit() or (string.count('.') == 1):
        c += 1
        break
if c == 1:
    print("String contains number")
else:
    print("String does not contains number")

Enter input:-  hguhj78


String contains number


### 33. Convert the total string in to lower case. Without using lower() function.

In [14]:
string = input("Enter input:- ")
res = ""
for i in string:
    res += chr((ord(i))+32)
print(res)

Enter input:-  SIRI


siri


#### 34. Convert the total string in to upper case. Without using upper() function.

In [16]:
string = input("Enter input:- ")
res = ""
for i in string:
    res += chr(ord(i)-32)
print(res)

Enter input:-  apple


APPLE


### 35. Show the below menu to the user until and until user select quit and display corresponding os message
### Menu:
### 1. windows
### 2. Linux
### 3. Mac#
### 4. quit

### 36. take a string from the user and check contains at least one digit or not?

In [17]:
string = input("Enter input:- ")

for i in string:
    if i.isdigit():
        print("String contains atleast one digit")
        break
else:
    print("String does not contains digit")


Enter input:-  132fghjbn


String contains atleast one digit


### 37. take a string from the user and check contains at least one alphabets or not?

In [18]:
string = input("Enter input:- ")

for i in string:
    if i.isalpha():
        print("String contains atleast one alphabets")
        break
else:
    print("String does not contains atleat one alphabets")


Enter input:-  dfxchgh34567#$%^&


String contains atleast one alphabets


### 38. take a string from the user and check contains at least one chars or not?

In [19]:
string = input("Enter input:- ")

if len(string)>=1:
    print("String contains at least one chars")
    
else: 
    print("String does not contains at least one chars")


Enter input:-  gh


String contains at least one chars


### 39. take a string from the user and check contains at least one capital letter or not?

In [21]:
string = input("Enter input:- ")

for i in string:
    if i.isupper():
        print("String contains at least one capital letter")
        break
else:
    print("String does not contains at least one capital letter")



Enter input:-  fchgvbj


String does not contains at least one capital letter


### 40. take a string from the user and check contains at least one small letter or not?

In [22]:
string = input("Enter input:- ")

for i in string:
    if i.islower():
        print("String contains at least one lower letter")
        break
else:
    print("String does not contains at least one lower letter")


Enter input:-  ghHJBKN


String contains at least one lower letter


### 41. Print the first 100 odd numbers

In [23]:
c = 0
res = ""
odd = 1
while 100 > c:
    if odd%2 != 0:
        res += str(odd)+" "
        c += 1
    odd += 1
        
print(res)      


1 3 5 7 9 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 51 53 55 57 59 61 63 65 67 69 71 73 75 77 79 81 83 85 87 89 91 93 95 97 99 101 103 105 107 109 111 113 115 117 119 121 123 125 127 129 131 133 135 137 139 141 143 145 147 149 151 153 155 157 159 161 163 165 167 169 171 173 175 177 179 181 183 185 187 189 191 193 195 197 199 


### 42. Determine the factors of a number entered by the user

In [24]:
user = int(input())
for i in range(1, user+1):
    if user%i == 0:
        print(i)
            

 18


1
2
3
6
9
18


### 43. Play a number guessing game (User enters a guess, you print YES or Higher or Lower). 
### This should continue until and until user gives a correct number or want to quit in the middle.
### Get a hidden number by using random.randint(1,100)

### 44. Take two numbers from the user a,b check whether a is divisible by b or not?

In [25]:
a=int(input("Enter any number: "))
b=int(input("Enter any number: "))

if a%b == 0:
    print("a is divisible by b")
else:
    print("a is not divisible by b")


Enter any number:  45
Enter any number:  76


a is not divisible by b


### 45. Find the sum of all the multiples of 3 or 5 below 1000

In [26]:
op = 0
for i in range(1,1000):
    if i%3==0 or i%5==0:
        print(i,end=" ")
        op += i
print()
print("The sum of all the multiples of 3 or 5 below 1000 is=",op)



3 5 6 9 10 12 15 18 20 21 24 25 27 30 33 35 36 39 40 42 45 48 50 51 54 55 57 60 63 65 66 69 70 72 75 78 80 81 84 85 87 90 93 95 96 99 100 102 105 108 110 111 114 115 117 120 123 125 126 129 130 132 135 138 140 141 144 145 147 150 153 155 156 159 160 162 165 168 170 171 174 175 177 180 183 185 186 189 190 192 195 198 200 201 204 205 207 210 213 215 216 219 220 222 225 228 230 231 234 235 237 240 243 245 246 249 250 252 255 258 260 261 264 265 267 270 273 275 276 279 280 282 285 288 290 291 294 295 297 300 303 305 306 309 310 312 315 318 320 321 324 325 327 330 333 335 336 339 340 342 345 348 350 351 354 355 357 360 363 365 366 369 370 372 375 378 380 381 384 385 387 390 393 395 396 399 400 402 405 408 410 411 414 415 417 420 423 425 426 429 430 432 435 438 440 441 444 445 447 450 453 455 456 459 460 462 465 468 470 471 474 475 477 480 483 485 486 489 490 492 495 498 500 501 504 505 507 510 513 515 516 519 520 522 525 528 530 531 534 535 537 540 543 545 546 549 550 552 555 558 560 561 56

### 46. Write a program to find out big of two numbers

In [27]:
a=int(input("Enter any number: "))
b=int(input("Enter any number: "))
if a>b:
    print("{} is big".format(a))
else:
    print("{} is big".format(b))

Enter any number:  15
Enter any number:  67


67 is big


### 47. Write a program to find out biggest number in the given numbers.

In [28]:
a=[1, 2, 5, 33, 6, 70, 9, 0, 3, 4]
ma=0
mi = a[0]
for i in a:
    if int(i)>ma:
        ma=i
print("The biggest number in the given numbers is=",ma)
for i in a:
    if int(i)<mi:
        mi=i
print("The smallest number in the given numbers is=",mi)



The biggest number in the given numbers is= 70
The smallest number in the given numbers is= 0


### 48. find out the index of third occurrence of given substring

In [29]:
s = input("Enter any string: ")
n = 3
print("{}rd occurance of the string={}".format(n,s[n-1]))


Enter any string:  HGFCGVHB


3rd occurance of the string=F


### 49. find out the index nth occurrence of given substring

In [32]:

s = input("Enter any string: ")
n=int(input("Enter index number: "))
print("{} occurance of the string is={}".format(n,s[n-1]))



Enter any string:  hello
Enter index number:  3


3 occurance of the string is=l


### 51. print the number in proper mathematical way.
### Consider that we have 6 digit numbers.
### Number format WAP> 10 -> 000010
### 100 -> 000100
### 1000 -> 001000
### 2345678 -> 2345678
### If the number has more than 6 digits then print as it is.

In [1]:
number = input("Enter number: ")
if len(number) >6:
    print(number)
else:
    print("%06d" % int(number))

Enter number:  65


000065


### 50. Take some single digit numbers from the user and findout min, maximum, sum, average

In [2]:
numbers = input("Enter numbers separated by spaces: ").split()
numbers = [int(num) for num in numbers if len(num) == 1]

if not numbers:
    print("No valid single-digit numbers were entered.")
else:
    minimum = numbers[0]
    maximum = numbers[0]
    sum = 0
    for num in numbers:
        if num < minimum:
            minimum = num
        if num > maximum:
            maximum = num
        sum += num

    average = sum / len(numbers)

    print("Minimum:", minimum)
    print("Maximum:", maximum)
    print("Sum:", sum)
    print("Average:", average)

Enter numbers separated by spaces:  2


Minimum: 2
Maximum: 2
Sum: 2
Average: 2.0


### 52. names ="emp1,emp2,emp3,emp4" iterate through the employee names.

In [3]:
names = "emp1,emp2,emp3,emp4"

# Split the string into a list of employee names
employee_names = names.split(",")

# Iterate through the employee names
for name in employee_names:
    print(name)
    

emp1
emp2
emp3
emp4


### 53. Take actual string, source string, destination string. replce first nth occurrences of source string with destination string of actual string.

In [6]:
# Taking inputs from the user
actual_string = input("Enter the actual string: ")
source_string = input("Enter the source string to be replaced: ")
destination_string = input("Enter the destination string: ")
n = int(input("Enter the number of occurrences to replace: "))

# Split the actual string by the source string
split_string = actual_string.split(source_string)

# Initialize an empty result string
result = ""

# Rebuild the string with replacements for the first n occurrences
for i in range(len(split_string) - 1):
    if i < n:
        result += split_string[i] + destination_string
    else:
        result += split_string[i] + source_string

# Add the last part of the string
result += split_string[-1]

# Output the final string
print("Modified string:", result)

Enter the actual string:  hello world siri
Enter the source string to be replaced:  siri
Enter the destination string:  hello
Enter the number of occurrences to replace:  5


Modified string: hello world hello


### 54. Take a two numbers from the user and do below menu driven operations

### 1. addition

### 2. multiples

### 3.division

### 4.sqrt

### 5. pow a**b

### 6.subtraction

### After selection do the corresponding operation.

### 
### Note: user may give int, or float numbers. You should check whether it is proper digits or not. I.e the user given string should be in the position to convert to float. Other wise show the “inproper string given” Error.

In [ ]:
# Taking two inputs from the user
num1 = input("Enter the first number: ")
num2 = input("Enter the second number: ")

# Checking if both inputs can be converted to float manually
if ('.' in num1 or num1.isdigit()) and ('.' in num2 or num2.isdigit()):
    num1 = float(num1)
    num2 = float(num2)

    # Displaying the menu
    print("\nMenu:")
    print("1. Addition")
    print("2. Multiplication")
    print("3. Division")
    print("4. Square Root")
    print("5. Power (a^b)")
    print("6. Subtraction")
    print("7. Quit")

    # Taking the user's choice
    choice = input("Select an operation (1-7): ")

    # If statement for each menu option
    if choice == '1':
        print(f"Addition: {num1} + {num2} = {num1 + num2}")

    elif choice == '2':
        print(f"Multiplication: {num1} * {num2} = {num1 * num2}")

    elif choice == '3':
        if num2 != 0:
            print(f"Division: {num1} / {num2} = {num1 / num2}")
        else:
            print("Error: Division by zero is not allowed.")

    elif choice == '4':
        from math import sqrt
        if num1 >= 0:
            print(f"Square Root of {num1} is {sqrt(num1)}")
        else:
            print("Error: Square root of negative numbers is not defined.")

        if num2 >= 0:
            print(f"Square Root of {num2} is {sqrt(num2)}")
        else:
            print("Error: Square root of negative numbers is not defined.")

    elif choice == '5':
        print(f"Power: {num1}^{num2} = {num1 ** num2}")

    elif choice == '6':
        print(f"Subtraction: {num1} - {num2} = {num1 - num2}")

    elif choice == '7':
        print("Exiting...")

    else:
        print("Invalid choice. Please select a valid option.")

else:
    print("Error: Improper string given. Please enter valid numbers.")

### 55. Take numbers from the user and find out min, maximum, sum, average

In [14]:
n =range(10)
for i in n:
    print(i)

0
1
2
3
4
5
6
7
8
9


In [15]:
n = int(input('enter the number of numbers:'))#2
max = None
min = None
add = 0
avg = 0
for i in range(n):# i in 0,1
    number = float(input(f"Enter number{i+1}"))# number1 =23,# number2 = 34
    add = add+number# 0+23 = 23, 23+34=57
    avg = add/n# 23/2 = 11.5, 57/2 = 28.5
    if max == None or max < number:#23<34, True
        max = number# 23, #34
    if min == None or min > number:# 23>34: false
        min = number#23
print("maximum number = ",max)
print("minimum number = ",min)
print("sum of all numbers = ",add)
print("average = ",avg)

enter the number of numbers: 3
Enter number1 12
Enter number2 23
Enter number3 34


maximum number =  34.0
minimum number =  12.0
sum of all numbers =  69.0
average =  23.0


### 56. l=[1,2,3,5,7,8,9,10,11,12,13,20,22,23,24,25,26,27,20,21,22,4] find out how many even numbers are there and how many odd numbers are there and how many positive numbers are there and how many negative numbers are there and how many prime numbers are there and how many perfect numbers are there and how many Armstrong numbers are there and how many palindrome numbers are there.

In [1]:
l = list(map(int,input().split()))
even_numbers = [i for i in l if i%2==0 ]
odd_numbers = [i for i in l if i%2==1]
positive_numbers = [i for i in l if i>0]
negative_numbers = [i for i in l if i<0]
def check_prime_numbers(l):
    prime_count = 0
    for i in l:
        count = 0 
        for j in range(2,i):
            if i%j == 0 :
                count += 1
        if count == 0 :
            prime_count += 1
    return prime_count
def count_perfect_numbers(l):
    perfect_number_count = 0
    for i in l:
        sum_prime = 0
        count = 0
        for j in range(2,i):
            if i%j ==0:
                count += 1
        if count == 0:
            sum_prime += 1
        if sum_prime == i:
            perfect_number_count +=1
    return perfect_number_count
print(len(even_numbers))
print(len(odd_numbers))
print(len(positive_numbers))
print(len(negative_numbers))
print(check_prime_numbers(l))
print(count_perfect_numbers(l))    


 68


### 57. Take a string from the user and find out how many digits are there, how many special symbols are there, how many small letters are there, how many caps are there. 

In [6]:
userinput = input("Enter a string : ")
digit_count = 0
symbol_count = 0
small_letter = 0
captial_count = 0
for i in user_input:
    if (i >= "a" and i <= "z"):
        small_letter += 1
    if (i >= "A" and i <= "Z"):
        captial_count += 1
    if (i >= "0" and i <= "9"):
        digit_count += 1
    else:
        symbol_count += 1
print("Digits:", digit_count)
print("Special symbols:", symbol_count)
print("Small letters:", small_letter)
print("Capital letters:", captial_count)
 

Enter a string :  !secfdbbfgb_98


Digits: 4
Special symbols: 10
Small letters: 7
Capital letters: 1


### 58. Take a char from the user and find out how many number of occurrences are there in given string 

In [1]:
user_input = input("Enter a string :")
user_input_1 = input("Enter a char :")
count = 0
for i in user_input:

  if i == user_input_1:
        count += 1
print("no of occurances are : ", count)
 

Enter a string : siri
Enter a char : i


no of occurances are :  2


In [2]:
user_input = input("Enter a string :")
user_input_1 = input("Enter a char :")count = 0
for i in user_input:
    if i == user_input_1:
        count += 1
print("no of occurances are : ", count)

Enter a string : srividya
Enter a char : i


no of occurances are :  2


### 59. Take a element from the user and find out how many times the element occurred in given list

In [3]:
def count_count():
    list_a = [1,2,3,4,5,2,5,2,6,1,2,5,2,1,5,4,3,8,9,2]
    user_input = input("Enter a number :")
    user_input = int(user_input)
    count = 0
    for i in list_a:
        if i == user_input:
            count += 1
    print("no of  occurances are ", count)


count_count()


Enter a number : 12121212121


no of  occurances are  0


### 60. Take an element from the user and find out how many number of occurrences are there in given tuple

In [5]:
def count_count():
    list_a = (1,2,3,4,5,2,5,2,6,1,2,5,2,1,5,4,3,8,9,2)
    user_input = input("Enter a number :")
    user_input = int(user_input)
    count = 0
    for i in list_a:
        if i == user_input:
            count += 1
    print("no of  occurances are ", count)

count_count()


Enter a number : 20


no of  occurances are  0


### 61. Reverse the string without effecting the special symbols. It involves three variations. Write code for three variations.

In [6]:
s = "abc123,#$45def6%$^789$%^"
print(s)
string_reverse = s[::-1]

res = ""
sub = ""
for i in string_reverse:
    if i.isalnum():
        res += sub + i
        sub = ""
    else:
        sub = i + sub
    

print(res)

print("$%^987%$^6fed54,#$321cba", "required output" ) 

abc123,#$45def6%$^789$%^
$%^987%$^6fed54,#$321cba
$%^987%$^6fed54,#$321cba required output


### 62.Input: abc123,#$45def6%$^789$%^, output: $%^987%$^6fed54,#$321cba

In [7]:
s      = "abc123,#$45def6%$^789$%^"
output = "9876fe,#$d54321%$^cba$%^"



### 63.Input: abc123,#$45def6%$^789$%^, output: 9876fe,#$d54321%$^cba$%^

### 64.Inout: "123,#$456%$^789$%^", Output: 321,#$654%$^987$%^

### Only numbers has to reverse

# FUNCTIONS: 

### 65. define a function to take person details name and age are mandatory parameters and height weight are optional parameters. If the user willing to pass any other details(like adhar, cell, pan, passport etc..) regarding him then your function should access those details. 

In [8]:
def person_details(name, age, height = None, weight = None, **kwargs):
    print(f"Name of the Person {name}")
    print(f"Age of the Person {age}")
    if height:
        print(f"Height of the Person :  {height}")
    if weight:
        print(f"Weight of the Person :  {weight}")
    if kwargs:
        for key, value in kwargs:
            print(f"{key} : {value}")

name = input("Enter The Name of the Person : ")
age = input("Enter an Age  of the Person : ")
height =  input("Enter height  of the Person : ")
weight =  input("Enter weight  of the Person : ")
aadhar =  input("Enter aadhar no.of the Person : ")
mobile_no =  input("Enter Mobile no.of the Person : ")

person_details(name, age, weight= weight)


Enter The Name of the Person :  10
Enter an Age  of the Person :  24
Enter height  of the Person :  5.4
Enter weight  of the Person :  65
Enter aadhar no.of the Person :  1243253856347567
Enter Mobile no.of the Person :  1234567898


Name of the Person 10
Age of the Person 24
Weight of the Person :  65


### 65a. rewrite above assignments by functions. Can use string functions to solve the string related assignments 

### 65 b. write a function to check given value is even or not 

In [9]:
def even_odd(a):
    if a %2 == 0:
        print(f"Given value {a} is Even")
    else:
        print(f"Given value {a} is Odd")

user_input = input("Enter a number to chexk even or not : " )
user_input = float(user_input)
even_odd(user_input)


Enter a number to chexk even or not :  2


Given value 2.0 is Even


### 65 c. write a function to check given value is prime or not 

In [10]:
def prime_no(check_number):
    count = 0
    for i in range(2, check_number+1):
        if check_number % i == 0 :
            count += 1
    return count

check_number = input("Enter a Number : ")
check_number = int(check_number)

count = prime_no(check_number)        
if count == 1:
    print(check_number, "is a Prime Number")
else:
    print(check_number, "is not a Prime Number")


Enter a Number :  7


7 is a Prime Number


### 65 d. write a function to check given 2 values are  divisible or not 

In [11]:
def div(a, b):
    print(a % b)
    if a % b == 0:
        print(" a is divisible by b ")
    else:
        print(" a is not divisible by b")
a = input("enter a  number :" )
a = int(a)
b = input("enter a number : ")
b = int(a)

div(a, b)


enter a  number : 10
enter a number :  8


0
 a is divisible by b 


# STRINGS 

### 67. take a string from the user and check contains only digits or not? 

In [12]:
def check_digits(user_input):
    count = 0
    for i in user_input:
        if not(i.isdigit()):
            print("contains other")
            break
        else:
            count += 1
    return count 
user_input = input("Enter a word with digits : ")
count = check_digits(user_input)
if count == len(user_input):
    print("contains only digits")


Enter a word with digits :  124


contains only digits


### 68. take a string from the user and check contains only alphabets or not? 

In [13]:
def check_alpha(user_input):
    count = 0
    for i in user_input:
        if not(i.isalpha()) :
            print("contains others")
            break
        else:
            count += 1
    return count
user_input = input("Enter a word with alphabets : ")
count = check_alpha(user_input)
if count == len(user_input):
    print("contains only alphabets")


Enter a word with alphabets :  word12


contains others


### 69. take a string from the user and check contains only special chars or not? 

In [14]:
def check_spl(user_input):
    count = 0
    for i in user_input:
        if i.isalnum()  :
            print("contains others")
            break
        else:
            count += 1
    return count
    
user_input = input("Enter a word with Special chars : ")
count = check_spl(user_input)

if count == len(user_input):
    print("contains only Special chars  ")


Enter a word with Special chars :  sxdss@


contains others


### 70. take a string from the user and check contains only capital letters or not? 

In [15]:
def check_capital(user_input):
    count = 0
    for i in user_input:
        if not(i.isupper()) :
            print("contains others")
            
            break
        else:
            count += 1
    return count
user_input = input("Enter a word with alphabets : ")
count = check_capital(user_input)
if count == len(user_input):
    print("contains only Capital Letters")


Enter a word with alphabets :  sfsdgfdfg


contains others


### 71. take a string from the user and check if it contains only small letters or not? 

In [16]:
def check_small(user_input):
    count = 0
    for i in user_input:
        if not(i.islower()) :
            print("contains others")
            break
        else:
            count += 1
    return count
user_input = input("Enter a word with alphabets : ")
count = check_small(user_input)
if count == len(user_input):
    print("contains only small letters")


Enter a word with alphabets :  fdrgtrhytjyk


contains only small letters


###  72. WAP to replace last n occurrence. 

In [17]:
def replace_str(list_a, source):
    n = 2
    count = 0
    for i in range(0, len(list_a)):
        if list_a[i] == source :
            list_a[i] = "APPLE"
            count += 1
            if count == n:
                break
    return list_a
a = "apple,orange,apple,grape,orange,apple,apple,orange"
print(a)
list_a = a.split(',')
list_a = list_a[::-1]
source = "apple"
list_a = replace_str(list_a, source)
print(",".join(list_a[::-1]))


apple,orange,apple,grape,orange,apple,apple,orange
apple,orange,apple,grape,orange,APPLE,APPLE,orange


### 73. WAP to check given string contains numbers or not. it should consider float numbers also. 

In [18]:
def check_numbers(user_input):
    for i in user_input:
        if (i.isdecimal()):
            print("contains numbers ")
            break
    else:
        print("not contains numbers ")

user_input = input("Enter a string it contains numbers and float values : " )
check_numbers(user_input)

Enter a string it contains numbers and float values :  "hai,1.2"


contains numbers 


### 74. Convert the total string in to lower case. 

### Convert the total string in to upper case. 

In [19]:
def str_upper(user_input):
    user_input = user_input.upper()
    return user_input

def str_lower(user_input):
    user_input = user_input.lower()
    return user_input
    
user_input = input("Enter a word with alphabets : ")
user_input1 = str_upper(user_input)
print("".join(user_input1))
user_input1 = str_lower(user_input)
print("".join(user_input1))


Enter a word with alphabets :  sadfghjnsqkjw


SADFGHJNSQKJW
sadfghjnsqkjw


### 75. Convert every word start letter into caps. 
### Some how title not working if it contains numbers and special symbols in the word 

In [20]:
def capitalize_words(text):
  words = text.split()
  capitalized_words = []
  for word in words:
    if word:
      capitalized_words.append(word.capitalize())
    else:
      capitalized_words.append(word)
  return " ".join(capitalized_words)

text = input("Enter a string : ")     # "hello world 123! special_symbol"
capitalized_text = capitalize_words(text)
print(capitalized_text)


Enter a string :  "hello"


"hello"


###  76.replace last two occurrences of given source string with destination string 
### preserve the delimiter after split. 

In [21]:
def replace_last_two_occurrences(text, source, destination, delimiter=' '):
  words = text.split(delimiter)
  count = 0
  new_words = []
  for i in range(len(words) - 1, -1, -1):
    if words[i] == source and count < 2:
      new_words.insert(0, destination)
      count += 1
    else:
      new_words.insert(0, words[i])
  return delimiter.join(new_words)


text = input("Enter a string : ") 
source = input("Enter a word : ") 
destination = input("Enter a word : ") 
new_text = replace_last_two_occurrences(text, source, destination)
print(new_text)


Enter a string :  sirisrisiri
Enter a word :  siri
Enter a word :  sri


sirisrisiri


### 77. write a program to check given substring is there in actual string or not? (search should be case insensitive) 
### example: act="python is a pure object oriented programing language" 

In [23]:
def contains_substring_case_insensitive(actual_string, substring):
    if substring.casefold() in actual_string.casefold():
        print(f"The substring '{substring_to_check}' is present in the string.")
    else:
        print(f"The substring '{substring_to_check}' is not present in the string.")

act = "python is a pure object oriented programing language"
substring_to_check = input("Enter a string : ")
contains_substring_case_insensitive(act, substring_to_check)


Enter a string :  srividya


The substring 'srividya' is not present in the string.


### 78.check whether “pure” is there in act or not. 
### Note: Use in operator 

In [24]:
def contains_substring_case_insensitive(actual_string, substring):
    if substring.casefold() in actual_string.casefold():
        print(f"The substring '{substring_to_check}' is present in the string.")
    else:
        print(f"The substring '{substring_to_check}' is not present in the string.")

act = "python is a pure object oriented programing language"
substring_to_check = "pure"
contains_substring_case_insensitive(act, substring_to_check)


The substring 'pure' is present in the string.


# DATA STRUCTURES  


### 79. l=[10,20,30,[40,50,60],70,[80,90,20]]. Convert this list as single dimensional list 

In [27]:
def dimensional_list(lst):
  flat_lst = []
  for item in lst:
    if type(item) == list :
      flat_lst.extend(flatten_list(item))
    else:
      flat_lst.append(item)
  return flat_lst

l = [10, 20, 30, [40, 50, 60], 70, [80, 90, 20]]
flat_l = dimensional_list(l)
print(flat_l)


NameError: name 'flatten_list' is not defined

### 80. input: "Google" print count of each character 

In [28]:
def char_count(input_string):
  char_counts = {}
  for char in input_string:
    if char in char_counts:
      char_counts[char] += 1
    else:
      char_counts[char] = 1
  return char_counts

input_string = "Google"
char_counts = char_count(input_string)
print(char_counts)


{'G': 1, 'o': 2, 'g': 1, 'l': 1, 'e': 1}


### 81. Convert n dimensional list to single dimensional list. 

In [29]:
def dimensional_list(lst):
  flat_lst = []
  for item in lst:
    if isinstance(item, list):
      flat_lst.extend(dimensional_list(item))
    else:
      flat_lst.append(item)
  return flat_lst


my_list = [[1, 2, [3, [1,2,3],4]], [5, 6], 7]
flat_list = dimensional_list(my_list)
print(flat_list)


[1, 2, 3, 1, 2, 3, 4, 5, 6, 7]


### 82. l=[1,2,3] just make it as a string. 

In [30]:
l = [1, 2, 3]
s = ''.join(map(str, l))
print(s)
print(list(s))


123
['1', '2', '3']


### 83. l=[1,2,3,[4,5,6],7,[8,9,10]] for single dimensional list 

In [31]:

def dimensional_list(lst):
  flat_lst = []
  for item in lst:
    if isinstance(item, list):
      flat_lst.extend(dimensional_list(item))
    else:
      flat_lst.append(item)
  return flat_lst


my_list = [1,2,3,[4,5,6],7,[8,9,10]]
flat_list = dimensional_list(my_list)
print(flat_list)


[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]


### 84. l=['a','A','b','B','d','D','c','C'] WAP to find out case insensitive count. 

In [32]:
def ele_count():
    a = {}
    for i in l:
        if i.casefold() in a:
            a[i.casefold()] += 1
        else:
            a[i.casefold()] = 1
    print(a)

l=['a','A','b','B','d','D','c','C'] 
ele_count()


{'a': 2, 'b': 2, 'd': 2, 'c': 2}


### 85. case insensitive search for an element

In [33]:
def ele_search():
    user_input = input("Enter a letter :")
    for i in l:
        if user_input.casefold() == i.casefold():
            print("Given element present in list")
            break
    else:
        print("Given element not present in list")
l=['a','A','b','B','d','D','c','C']
ele_search()


Enter a letter : s


Given element not present in list


### 86. l=['a','A','b','B','d','D','c','C']  sort the list properly 

In [34]:

l=['a','A','b','B','d','D','c','C']
l.sort()
print(l)


['A', 'B', 'C', 'D', 'a', 'b', 'c', 'd']


### 87. find the start position of the largest block of repeated characters in a given string 

In [35]:
def find_largest_block_start(text):
    if not text:
        return -1
    max_length = 0
    start_position = -1
    current_length = 1
    current_start = 0
    
    for i in range(1, len(text)):
        if text[i] == text[i - 1]:
            current_length += 1
        else:
            if current_length > max_length:
                max_length = current_length
                start_position = current_start
            current_length = 1
            current_start = i
    if current_length > max_length:
        max_length = current_length
        start_position = current_start
    return start_position

text = "aaabbbcccdddd"
start = find_largest_block_start(text)
if start != -1:
  print(f"The largest block of repeated characters starts at position {start}")
else:
  print("No repeated characters found.")


The largest block of repeated characters starts at position 9


### 88. WAP to find union and intersection of lists. 

In [36]:
def find_union_intersection(list1, list2):
  union = list(set(list1) | set(list2))
  intersection = list(set(list1) & set(list2))
  return union, intersection

list1 = [1, 2, 3, 4, 5]
list2 = [3, 4, 5, 6, 7]
union, intersection = find_union_intersection(list1, list2)

print(f"Union: {union}")
print(f"Intersection: {intersection}")


Union: [1, 2, 3, 4, 5, 6, 7]
Intersection: [3, 4, 5]


### 89. input: fun(5) output: [1,2,3,4,3,2,1] 

In [37]:
def fun(n):
  result = []
  for i in range(1, n):
    result.append(i)
  for i in range(n - 2, 0, -1):
    result.append(i)
  print(result)

fun(5)


[1, 2, 3, 4, 3, 2, 1]


### 90. input fun('abc') output: [[],][a],[b],[c],[a,b],[b,c],[c,a],[a,b,c]] 

In [38]:

def fun(s):
  result = [[]]
  for i in range(len(s)):
    result.append([s[i]])
  for i in range(len(s) - 1):
    result.append([s[i], s[i + 1]])
  for i in range(len(s) - 2):
    result.append([s[i], s[i + 1], s[i + 2]])
  return result

print(fun('abc'))


[[], ['a'], ['b'], ['c'], ['a', 'b'], ['b', 'c'], ['a', 'b', 'c']]


### 91. Remove duplicates from the list: a=[1,2,3,2,3,4,1,,3,4] 

In [39]:
a = [1, 2, 3, 2, 3, 4, 1, 3, 4]
new_list = []
for i in a:
  if i not in new_list:
    new_list.append(i)
print(new_list)


[1, 2, 3, 4]


### 92. l=['1','2','3'] get the sum of the list 

In [40]:
def sum_of(l):
    sum_ = 0
    for i in l:
        sum_ += int(i)
    print(sum)

l = ["1", "2", "3"]
sum_of(l)


<built-in function sum>


### 93. l1=[1,2,3,4] l2=[5,6,7,8] sum of two lists 

In [41]:
def sum_of(l1, l2):
    sum_ = 0
    l1.extend(l2)
    for i in l1:
        sum_ += i
    print(sum_)

l1=[1,2,3,4]
l2=[5,6,7,8]
sum_of(l1, l2)


36


### 94. Find third max value of element in a list with soring and without sorting a list. 

### 95. Input = ["1/1","1/2","1/3","1/4","2/5","2/6","2/8"] Output = [['1/1-4'], ['2/5-6'], ['2/8']] 

### 96. l=[1,2,3,5,7,8,9,10,11,12,13,20,22,23,24,25,26,27,20,21,22,4] 
### output = [[1, 2, 3], [5], [7, 8, 9, 10, 11, 12, 13], [20], [22, 23, 24, 25, 26, 27], [20, 21, 22], [4]] 

### 97. input = 1,2,3,4,5,6,8,10 output = odd,even,odd,even,odd,even,even,even 

In [42]:
def fun(l):
    for i in l:
        if i % 2 == 0:
            print("even", end=",")
        else:
            print("odd", end=",")

l =list(map(int, input("enter numbers :").split(",")))
fun(l)


enter numbers : 2,6,7,9


even,even,odd,odd,

### 98. input n=3 

    ###		output:     111 

    ###    		101 

     ###   		111 

In [44]:
def fun(l):
    for i in range(1, l+1):
        if i % 2 == 0:
            print("101")
        else:
            print("111")

l =input("enter number :")
l = int(l)
fun(l)


enter number : 3


111
101
111


### 99. input: Google 
###     output: {'g':2,'o':2,'l':1,'e':1} use dictionary comprehension 

In [45]:
def fun(user_input):
    dict_a = {}
    for i in user_input:
        if i.casefold() in dict_a:
            dict_a[i.casefold()] += 1
        else:
            dict_a[i.casefold()] = 1
    print(dict_a)
user_input = input("Enter a String :")
fun(user_input)


Enter a String : hai


{'h': 1, 'a': 1, 'i': 1}


### 100. keys=['k1','k2'], values = ['v1','v2'] form a dictionary. 

In [46]:
def fun():
    keys = ['k1', 'k2']
    values = ['v1', 'v2']
    my_dict = dict(zip(keys, values))
    print(my_dict)

fun()


{'k1': 'v1', 'k2': 'v2'}


### 101. Sort the list marks = [("mohan", 80), ("satish", 90), ("purnesh", 40), ("venkat", 30)]
### according to descending order of marks


### 102.write a function to get dynamic list for floating numbers also based on strat and end and step parameters


### 103. find out all perfect numbers in given range


In [47]:
def perfect_no(n):
    res = []
    for i in range(1, n+1):
        sum_of = 0
        for j in range(1, i):
            if i % j == 0:
                sum_of += j
        if sum_of == i:
            res.append(i)
    print(res)

n = input("Enter a number :")
if (int(n)>0):
    n = int(n)
    perfect_no(n)
else:
    print("Give Positive number")

Enter a number : 3


[]


### 104. WAP to do all stack operations using lists


### 105.WAP to do all queue operations using lists


### 106. WAP to remove n occurrences of specified element from a list


In [48]:
def remove_n_occurances():
    list_a = "AJA, consulting, Services, LLP"
    n = 2
    a = list_a.replace(",", "@", n)
    print(a)

remove_n_occurances()

AJA@ consulting@ Services, LLP


### 107. compare two lists ignore order. i.e return True l1=[1,2,3,4],l2=[4,2,3,1], fun(l1,l2)-> True


### 108. XOR operation in python.


### 109. how to remove all occurrences of the given element in a list


### 110. how to remove first n occurrences of the given element in a list


### 111. how to remove last n occurrences of the given element in a list


### 112. how to remove nth occurrences of the given element in a list


### 113. WAP to generate list of floats i.e: fun(0,1,0.1), [0,0.1,0.2,0.3,0.4,0.5,0.6,0.7,0.8,0.9]


### 114. WAP to remove all occurrences of given substring from actual string


### 115. WAP to remove first n occurrences of given substring from actual string


### 116. WAP to remove last n occurrences of given substring from actual string


### 117. WAP to replace last n occurrences of given substring with destination string in actual string


### 118. WAP to sort the string.


### 119. take a coma separated numbers and find out max number.
